### PR TITLE
Fb takyoon swap admin

### PR DESCRIPTION
**Pull request by takYoon**

### CODE REVIEW

File: packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
+export interface IConversionOrder {
```
Comment (vince8x) [inline packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts:1]: The interface is not extends by the ConversionOrder class?

File: packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  amount: Amount;
+  isMintBaton: boolean;
+}
+
+export interface SlpBurn {
+  token: Token;
+  tokenId: string;
+}
+
+export interface Amount2 {
```
Comment (vince8x) [inline packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts:82]: What is the purpose of Amount 2? Does it seem to be exactly the same as Amount?

File: packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  unsigned: boolean;
+}
+
+export interface Amount3 {
+  low: number;
+  high: number;
+  unsigned: boolean;
+}
+
+export interface SlpToken2 {
```
Comment (vince8x) [inline packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts:115]: SlpToken2 and SlpToken look the same

File: packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  prevOut: PrevOut;
+  inputScript: string;
+  outputScript: string;
+  value: Value;
+  sequenceNo: any;
+  slpBurn: SlpBurn;
+  slpToken: SlpToken;
+}
+
+export interface Value2 {
```
Comment (vince8x) [inline packages/bitcore-wallet-service/src/lib/model/conversionOrder.ts:103]: Value2 and Value look the same

File: packages/bitcore-wallet-service/src/lib/model/order.ts [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  coinConfig?: CoinConfig;
+  toTokenInfo?: TokenInfo;
+  fromTokenInfo?: TokenInfo;
+  note?: string;
+  pendingReason?: string;
+  lastModified?: number;
+  isResolve?: boolean;
+}
+
+export class Order {
```
Comment (vince8x) [inline packages/bitcore-wallet-service/src/lib/model/order.ts:41]: The class Order should extend the interface IOrder?

File: packages/bitcore-wallet-service/src/config.ts [hunk 6]
```diff
@@ -51,7 +51,7 @@ module.exports = {
     },
     xec: {
       livenet: {
-        url: 'http://127.0.0.1:3000'
+        url: 'http://dev.abcpay.cash:3000'
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/config.ts:54]: will revert this file

File: packages/bitcore-wallet-service/src/config.ts [hunk 7]
```diff
@@ -18,7 +18,7 @@ module.exports = {
   // CAinter1: '', // ex. 'COMODORSADomainValidationSecureServerCA.crt'
   // CAinter2: '', // ex. 'COMODORSAAddTrustCA.crt'
   // CAroot: '', // ex. 'AddTrustExternalCARoot.crt'
-
+  sharedKey: 'ajkhjgkloiuyrjkds',
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/config.ts:21]: will revert this file

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 8]
```diff
@@ -1296,6 +1335,67 @@ export class ExpressApp {
...
+      } catch (err) {
+        if (err) return returnError(err, res, req);
+      }
+      server.getAllTokenInfo((err, tokenInfoList) => {
+        if (err) returnError(err, res, req);
+        res.json(tokenInfoList);
+      });
+    });
+
+    router.post('/v3/sendToken/', (req, res) => {
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/lib/expressapp.ts:1370]: will make this later

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 9]
```diff
@@ -1315,15 +1415,446 @@ export class ExpressApp {
...
+      } catch (ex) {
+        return returnError(ex, res, req);
+      }
+      server.checkConversion((err, order) => {
+        if (err) return returnError(err, res, req);
+        res.json(order);
+      });
+    });
+
+    router.post('/v3/order/update', (req, res) => {
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/lib/expressapp.ts:1780]: will replace by PUT

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 10]
```diff
@@ -1315,15 +1415,446 @@ export class ExpressApp {
...
+    });
+
+    router.post('/v3/coinconfig/filter', (req, res) => {
+      let server;
+      try {
+        server = getServer(req, res);
+      } catch (ex) {
+        return returnError(ex, res, req);
+      }
+      server.rescanWalletsInKeys((err, listCoinConfig) => {
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/lib/expressapp.ts:1852]: will have check for purpose

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 11]
```diff
@@ -1765,6 +2207,62 @@ export class WalletService {
...
+      } catch (ex) {
+        return cb(null, []);
+      }
+      if (addrObj.network.name != wallet.network) {
+        return cb(null, []);
+      }
+
+      this._getBlockchainHeight(wallet.coin, wallet.network, (err, height, hash) => {
+        if (err) return cb(err);
+        bc.getAddressUtxos(address, height, wallet.coin, async (err, allUtxos) => {
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/lib/server.ts:2237]: skip

File: packages/bitcore-wallet-service/src/lib/model/tokenInfo.ts [hunk 12]
```diff
@@ -16,3 +16,10 @@ export interface TokenInfo {
   totalMinted: number;
   versionType: number;
 }
+
+export interface TokenItem {
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/lib/model/tokenInfo.ts:20]: will seperate into new ticket

File: packages/bitcore-wallet-service/src/lib/model/keys.ts [hunk 13]
```diff
@@ -0,0 +1,16 @@
+interface Keys {
+  keyFund: string;
+  keyReceive: string;
+  createdOn: number;
+  lastModified: number;
+  hashPassword: string;
+  hashRecoveryKey: string;
+}
+
+interface KeysConversion {
```
[the remark this answers is not in the document]
Comment (takYoon) [reply packages/bitcore-wallet-service/src/lib/model/keys.ts:10]: will come in new ticket

**Pull request merged**